TodoApp.ConsoleApp/Services/TodoService.cs

```diff
@@ -1,77 +1,92 @@
 ﻿using System.Collections.Generic;
-using TodoApp.Library.Data;
-using TodoApp.Library.Models;
+using TodoApp.ConsoleApp.Repositories.Models;
+using TodoApp.ConsoleApp.Repositories.XmlRepository;
 using System.Linq;
 using System;
+using System.IO;
+using System.Xml;
+using System.Text;
 
 namespace TodoApp.ConsoleApp.Services
 {
     public class TodoService : ITodoService
     {
-        public IEnumerable<Task> GetAll()
-        {
-            XMLTaskReader reader = new XMLTaskReader();
+        static string Path { get; set; }
+        TodoRepository repo = new TodoRepository(Path);
 
-            IEnumerable<Task> tasks = reader.ReadTasks();
+        public TodoService(string path)
+        {
+            Path = path;
 
-            return tasks;
+            if (!File.Exists(Path))
+            {
+                XmlWriterSettings xmlWriterSettings = new XmlWriterSettings();
+                xmlWriterSettings.Indent = true;
+                Encoding encoding = Encoding.GetEncoding("UTF-8");
+                using (XmlWriter writer = XmlWriter.Create(Path))
+                {
+                    writer.WriteStartDocument();
+                    writer.WriteStartElement("todos");
+                    writer.WriteEndElement();
+                    writer.WriteEndDocument();
+                    writer.Close();
+                }
+            }
```

Comment:
TodoService do not care about the data, where it comes from, and where it is saved. It is not its responsibility to make sure that the file exists.
Tomorrow we might want to use JSON instead of XML. If that happens, the only change should happen in the Repository module!

Remove this from here. Make this validation in RepositoryBase (so every repository is validated).
Be consistent and use XDocument.
```cs
if (File.Exists(path))
{
    Document = XDocument.Load(path);
}
else
{
    Document = new XDocument(new XElement(containerName));
}
```